gamebook/chance_option.py

```diff
@@ -2,10 +2,13 @@
 from gamebook.option import *
 
 class ChanceOptions(Option):
-    """an option which the player won't necessarily be succesful in preforming(depends on the decision's success rate)"""
-    def DecisionDraw(self):
+    """an option which the player won't necessarily be succesful in preforming(depends on the nextscene's success rate)"""
+    def Evaluate_decision(self):
+        """
         wheights = []
-        for i in range(len(self.decision)-1):
-            wheights[i] = self.decision[i].get_success_rate()
-        draw = random.choices(self.decision,wheights,k=1)
-        return draw
+        for i in self.nextscene:
+            wheights.append(i.get_success_rate())
+        """
+        wheights = [i.get_success_rate() for i in self.nextscene]
```

Comment:
Did this solve the bug you had before? because I don't see length and index out of range checks here.

Reply:
yes i did solve it. there was a problem with the range() and len(), the code in the comment block passed the test

Reply:
Ok, remove the comment then, if it isn't needed.

Reply:
:+1: